terraform/modules/jenkins/ec2.tf

```diff
@@ -91,5 +91,5 @@ data "aws_iam_policy_document" "lambda_policy" {
 resource "aws_network_interface" "ec2_network_interface" {
   count          = var.az_count
   subnet_id       = var.ecs_private_subnet[0]
-  security_groups = [aws_security_group.ec2.id]
+  security_groups = [aws_security_group.ec2_internal.id, aws_security_group.ec2_cloudfront_global.id, aws_security_group.ec2_cloudfront_regional.id]
```

Comment:
I don't know much about ENIs - are you supposed to add them to all these different security groups? Or should you give them their own security group set up ingress/egress rules to allow access from/to services with these security groups?

Reply:
I'm not sure what you mean but the reason they are separate is that there are a log of cloudfront edge IP ranges and only a certain number of rules you can have in a security group, so they're separated out. This is what AWS recommends,  if you look at the lambda code they provide for updating them, this is what it's expecting.